dist/src/main/java/io/camunda/application/commons/migration/MigrationsRunner.java

```diff
@@ -34,7 +35,26 @@ public MigrationsRunner(final List<Migrator> migrators) {
   @Override
   public void run(final ApplicationArguments args) throws Exception {
     LOG.info("Starting {} migration tasks", migrators.size());
-    final Exception migrationsExceptions = new Exception("migration is failed");
+    /* Detach from main Spring thread */
+    CompletableFuture.runAsync(
+            () -> {
+              try {
+                startMigrators(args);
+              } catch (final Exception e) {
+                throw new RuntimeException(e);
+              }
+            },
+            Executors.newSingleThreadExecutor())
```

Comment:
:x: Please set a meaningful thread name.
:wrench: I think a simple `new Thread(...).run()` might be preferable here.